### PR TITLE
[WEBDEV-5290] Fix stats size and spacing

### PR DESCRIPTION
**Pull request by dualcnhq**

### CODE REVIEW

File: src/styles/item-tile-styles.ts [hunk 1]
```diff
@@ -0,0 +1,131 @@
+import { css } from 'lit';
```
Comment (mc2) [inline src/styles/item-tile-styles.ts:1]: question: Why did you remove the styles?

It would make sense to me if multiple components used the same style file. But for 1-component corresponding to 1-style file, I think I, Drini, and Isa have a preference that they be in the same component file.

Comment (mc2) [reply src/styles/item-tile-styles.ts:1]: blocker: If you prefer the styles in a separate file, then at least move this file into `tiles/grid`, in the same folder as the component.

Kind of a Principle: 
- The contents of a component should be together, as much as possible.

Imagine if you were distributing the item-tile component -- as a component. You would not expect the component consumer, a page author, to create a separate, global`/styles` folder.

**Review by mc2 — CHANGES_REQUESTED** [review]

File: src/tiles/item-stats.ts [hunk 2]
```diff
@@ -0,0 +1,67 @@
+import { CSSResultGroup, html, LitElement } from 'lit';
```
Comment (mc2) [inline src/tiles/item-stats.ts:1]: Great idea moving this out as a separate component.

[the remark this answers is not in the document]
Comment (mc2) [reply src/tiles/item-stats.ts:1]: blocker: We want component names to be clear and accurate.

Since this component is part of both `account-tile` and `item-tile` it's not an "item" component, but a "tile" component.

Please rename to `tile-stats.ts`

File: src/styles/item-stats-styles.ts [hunk 3]
```diff
@@ -0,0 +1,40 @@
+import { css } from 'lit';
```
Comment (mc2) [inline src/styles/item-stats-styles.ts:1]: blocker: As below with `item-stats.ts` 

- placement next to sibling components 
- naming

Please move to `src/tiles/grid/` and rename to `tile-stats-styles.ts`

**Review by mc2 — CHANGES_REQUESTED** [review]

File: src/tiles/grid/tile-stats.ts [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+  @property({ type: Number }) viewCount?: number;
+
+  @property({ type: Number }) favCount?: number;
+
+  @property({ type: Number }) commentCount?: number;
+
+  render() {
+    console.log('tile-stats here');
+    return html`
+      <div class="stats-wrapper">
```
Comment (iisa) [inline src/tiles/grid/tile-stats.ts:26]:
```suggestion
      <div class="item-stats">
```
suggestion: create more meaningful class names

File: src/tiles/grid/tile-stats.ts [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+        <div id="stats-holder">
```
Comment (iisa) [inline src/tiles/grid/tile-stats.ts:27]:
```suggestion
        <div id="list">
```

File: src/tiles/grid/tile-stats.ts [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+
+  @property({ type: Number }) itemCount?: number;
+
+  @property({ type: Number }) viewCount?: number;
+
+  @property({ type: Number }) favCount?: number;
+
+  @property({ type: Number }) commentCount?: number;
+
+  render() {
```
Comment (iisa) [inline src/tiles/grid/tile-stats.ts:23]: 👍 nice & stateless

**Review by iisa — COMMENTED** [review]
@dualcnhq good job on the style fix and locking in the desired design.

looks good!
please can you add some unit tests against the new <tile-stats> plus other elements that have been updated?
also, I have some suggestions around possibly flattening the dom even further & an open question on how we can simplify some of these class names (remove `-holder|wrapper`) and use class names that are closer to what they are responsible for...

thanks! lmk what you think

File: src/tiles/grid/account-tile.ts [hunk 7]
```diff
@@ -14,43 +11,37 @@ export class AccountTile extends LitElement {
   render() {
     return html`
       <div class="outer-holder">
+        <div id="header-holder">
+          <div id="title-holder">
```
[the remark this answers is not in the document]
Comment (dualcnhq) [reply src/tiles/grid/account-tile.ts:15]: most of this were old code when we started working on it.
just didn't touch it at the moment when i added tweaks

File: src/tiles/grid/tile-stats.ts [hunk 8]
```diff
@@ -0,0 +1,104 @@
...
+    console.log('tile-stats here');
+    return html`
+      <div class="stats-wrapper">
+        <div id="stats-holder">
+          <div class="col">
+            <mediatype-icon
+              .mediatype=${this.mediatype}
+              style="--iconHeight:25px; --iconWidth:25px;"
```
[the remark this answers is not in the document]
Comment (dualcnhq) [reply src/tiles/grid/tile-stats.ts:31]: what do you mean here? @iisa

File: src/tiles/grid/account-tile.ts (hunk 7 above)
[the remark this answers is not in the document]
Comment (iisa) [reply src/tiles/grid/account-tile.ts:15]: Please do check if we can remove. Lesser nesting is preferred..

File: src/tiles/grid/tile-stats.ts (hunk 8 above)
[the remark this answers is not in the document]
Comment (iisa) [reply src/tiles/grid/tile-stats.ts:31]: Move that in-line style to css declaration

[the remark this answers is not in the document]
Comment (dualcnhq) [reply src/tiles/grid/tile-stats.ts:31]: if it's about the style props, it was intentionally a passed props to mediatype-icon component as we have dynamic sizing for the mediatype-icon component

[the remark this answers is not in the document]
Comment (bfalling) [reply src/tiles/grid/tile-stats.ts:31]: @dualcnhq: Passing in sizing values is fine, but passing in using a `style` attribute is a little suspicious. e.g. If you want to pass icon height and width, then would it make more sense to use explicit parameters for height and width?

File: src/tiles/grid/account-tile.ts (hunk 7 above)
[the remark this answers is not in the document]
Comment (nsharma123) [reply src/tiles/grid/account-tile.ts:15]: I also want to mentioned on more thing here on class name on different tiles. there is not similarity with class name even we are displaying stats icons/count on account and item tiles.

eg. item-tiles container has different HTML structure including classes name and account-tiles does have different structure even we are showing same content like stats icon and count.

[the remark this answers is not in the document]
Comment (nsharma123) [reply src/tiles/grid/account-tile.ts:15]: image

File: src/tiles/grid/account-tile.ts [hunk 9]
```diff
@@ -14,43 +11,37 @@ export class AccountTile extends LitElement {
...
-            <div class="stat-icon">
-              ${reviewsIcon}
-              <h3>${this.model?.commentCount}</h3>
+              <span>
+                Archivist since ${this.model?.dateAdded?.getFullYear()}
+              </span>
             </div>
           </div>
         </div>
+
```
Comment (nsharma123) [inline src/tiles/grid/account-tile.ts:37]: Thanks you !

File: src/tiles/grid/item-tile.ts [hunk 10]
```diff
@@ -19,91 +16,82 @@ export class ItemTile extends LitElement {
...
   }
 
   static get styles(): CSSResultGroup {
-    const cornerRadiusCss = css`var(--collectionTileCornerRadius, 4px)`;
-
     return css`
       #container {
         background-color: #ffffff;
-        border-radius: ${cornerRadiusCss};
+        border-radius: var(--collectionTileCornerRadius, 4px);
```
Comment (nsharma123) [inline src/tiles/grid/item-tile.ts:62]: as I remember, we don't want to use px unit in CSS instead we use rem units.

File: src/tiles/grid/tile-stats.ts (hunk 6 above)
Comment (nsharma123) [reply src/tiles/grid/tile-stats.ts:23]: Thanks you Mark!

File: src/tiles/grid/tile-stats.ts [hunk 11]
```diff
@@ -0,0 +1,104 @@
...
+            >
+            </mediatype-icon>
+          </div>
+          <div class="col">
```
Comment (nsharma123) [inline src/tiles/grid/tile-stats.ts:35]: Goody goody!

File: src/tiles/grid/tile-stats.ts [hunk 12]
```diff
@@ -0,0 +1,104 @@
...
+          </div>
+        </div>
+      </div>
+    `;
+  }
+
+  static get styles(): CSSResultGroup {
+    return css`
+      .stats-wrapper {
+        height: 35px;
```
Comment (nsharma123) [inline src/tiles/grid/tile-stats.ts:65]: Open question to everyone, are we now supporting using px unit instead of rem/em units? At PB side, we totally against it unless its really needed. @iisa @mc2 @dualcnhq

File: src/tiles/grid/item-tile.ts (hunk 10 above)
Comment (dualcnhq) [reply src/tiles/grid/item-tile.ts:62]: it was stated in the mockups though to use `px`

File: src/tiles/grid/tile-stats.ts (hunk 12 above)
Comment (dualcnhq) [reply src/tiles/grid/tile-stats.ts:65]: `px` was set in the mockups so i just followed it.

cc @pezvi

File: src/tiles/grid/item-tile.ts (hunk 10 above)
Comment (iisa) [reply src/tiles/grid/item-tile.ts:60]: @nsharma123 yep, for pixel perfect design parts like border-radius, we can use px. for text, rule of thumb is using rem.

File: src/tiles/grid/account-tile.ts (hunk 7 above)
[the remark this answers is not in the document]
Comment (iisa) [reply src/tiles/grid/account-tile.ts]: the `<tile-stats>` is a new component and looks great - encapsulated dom & style. 
the dom change is ok as it seems like we are moving to better patterns, creating scoped components, writing semantic dom ^_^

File: src/tiles/grid/tile-stats.ts (hunk 8 above)
[the remark this answers is not in the document]
Comment (iisa) [reply src/tiles/grid/tile-stats.ts]: what I meant was adding this to the css block as such

```css
mediatype-icon {
  --iconHeight: 25px;
  --iconWidth: 25px;
}
```

**Review by iisa — COMMENTED** [review]
this looks really good @dualcnhq!

in an encapsulated world like web components, we can actually be more precise with our class namings to reflect its scope.
i've got some "suggestions" added to show how we can better name the elements in `<item-tile>` and `<account-tile>`.  this really helps shape the DOM tree well as when one opens the inspector, it's clear what the responsibilities of each element is.

also, i see some nested elements with flexbox properties though their direct parent is not `display: flex` -> a sweep thru these properties to confirm whether or not we need them will help keep the components nice and tidy.

a unit test for the `<tile-stats>` display will be good to have + continue to provide "code security & peace of mind" to this repo. then adding tests to check for its existence in an item & account tile.

thanks for the good work 💪 you pared it down a lot since we talked last week.

File: src/tiles/grid/item-tile.ts [hunk 13]
```diff
@@ -19,91 +16,80 @@ export class ItemTile extends LitElement {
...
+        padding-left: 5px;
+        padding-right: 5px;
+      }
+
+      .title-wrapper {
+        flex-shrink: 0;
+      }
+
+      .image-wrapper {
+        flex-grow: 1;
```
[the remark this answers is not in the document]
Comment (dualcnhq) [reply src/tiles/grid/item-tile.ts:78]: yup got this from here: https://developer.mozilla.org/en-US/docs/Web/CSS/Layout_cookbook/Sticky_footers

File: src/tiles/grid/tile-stats.ts [hunk 14]
```diff
@@ -0,0 +1,104 @@
...
+  static get styles(): CSSResultGroup {
+    return css`
+      mediatype-icon {
+        --iconHeight: 25px;
+        --iconWidth: 25px;
+      }
+
+      .item-stats {
+        height: 35px;
+        flex-shrink: 0;
```
[the remark this answers is not in the document]
Comment (dualcnhq) [reply src/tiles/grid/tile-stats.ts:66]: got the idea from this: https://developer.mozilla.org/en-US/docs/Web/CSS/Layout_cookbook/Sticky_footers

File: src/tiles/grid/tile-stats.ts [hunk 15]
```diff
@@ -0,0 +1,104 @@
...
+            </p>
+          </div>
+        </div>
+      </div>
+    `;
+  }
+
+  static get styles(): CSSResultGroup {
+    return css`
+      mediatype-icon {
```
Comment (iisa) [inline src/tiles/grid/tile-stats.ts:59]: 👍

**Review by iisa — APPROVED** [review]
looks good, code-wise.

**Review by mc2 — APPROVED** [review]
Looks good to me
